src/graphnet/pisa/utils.py

```diff
@@ -0,0 +1,586 @@
+
+[flux.barr_simple]
+calc_mode = true_allsky_fine
+apply_mode = events
+param.nu_nubar_ratio = 1.0 +/- 0.1
+param.nu_nubar_ratio.fixed = True
+param.nu_nubar_ratio.range = nominal + [-3., +3.] * sigma
+param.nue_numu_ratio = 1.0 +/- 0.05
+param.nue_numu_ratio.fixed = False
+param.nue_numu_ratio.range = nominal + [-0.5, +0.5]
+param.Barr_uphor_ratio = 0.0 +/- 1.0
+param.Barr_uphor_ratio.fixed = False
+param.Barr_uphor_ratio.range = nominal + [-3.0, +3.0]
+param.Barr_nu_nubar_ratio = 0.0 +/- 1.0
+param.Barr_nu_nubar_ratio.fixed = False
+param.Barr_nu_nubar_ratio.range = nominal + [-3.0, +3.0]
+param.delta_index = 0.0 +/- 0.1
+param.delta_index.fixed = False
+param.delta_index.range = nominal + [-5, +5] * sigma
+
+[osc.prob3]
+calc_mode = true_allsky_fine
+apply_mode = events
+param.earth_model = osc/PREM_12layer.dat
+param.YeI = ${earth:YeI}
+param.YeM = ${earth:YeM}
+param.YeO = ${earth:YeO}
+param.detector_depth = ${earth:detector_depth}
+param.prop_height = ${earth:prop_height}
+param.theta12 = ${osc:theta12}
+param.theta12.fixed = True
+param.nh.theta13 = ${osc:theta13_nh}
+param.nh.theta13.fixed = False
+param.nh.theta13.range = ${osc:theta13_nh.range}
+param.ih.theta13 = ${osc:theta13_ih}
+param.ih.theta13.fixed = False
+param.ih.theta13.range = ${osc:theta13_ih.range}
+param.nh.theta23 = ${osc:theta23_nh}
+param.nh.theta23.fixed = False
+param.nh.theta23.range = ${osc:theta23_nh.range}
+param.nh.theta23.prior = uniform
+param.ih.theta23 = ${osc:theta23_ih}
+param.ih.theta23.fixed = False
+param.ih.theta23.range = ${osc:theta23_ih.range}
+param.ih.theta23.prior = uniform
+param.nh.deltacp = 0.0 * units.degree
+param.nh.deltacp.fixed = True
+param.nh.deltacp.range = ${osc:deltacp_nh.range}
+param.nh.deltacp.prior = uniform
+param.ih.deltacp = 0.0 * units.degree
+param.ih.deltacp.fixed = True
+param.deltam21 = ${osc:deltam21}
+param.deltam21.fixed = True
+param.nh.deltam31 = ${osc:deltam31_nh}
+param.nh.deltam31.fixed = False
+param.nh.deltam31.prior = uniform
+param.nh.deltam31.range = [0.001, +0.007] * units.eV**2
+param.ih.deltam31 = ${osc:deltam31_ih}
+param.ih.deltam31.fixed = False
+param.ih.deltam31.prior = uniform
+param.ih.deltam31.range = [-0.007, -0.001] * units.eV**2
+
+[aeff.aeff]
+calc_mode = events
+apply_mode = events
+param.livetime = 10 * units.common_year
+param.aeff_scale = 1.0
+param.aeff_scale.fixed = False
+param.aeff_scale.prior = uniform
+param.aeff_scale.range = [0.,3.] * units.dimensionless
+param.nutau_cc_norm = 1.0
+param.nutau_cc_norm.fixed = True
+param.nutau_cc_norm.range = [0.2, 2.0] * units.dimensionless
+param.nutau_cc_norm.prior = uniform
+param.nutau_norm = 1.0
+param.nutau_norm.fixed = False
+param.nutau_norm.range = [-1.0, 8.5] * units.dimensionless
+param.nutau_norm.prior = uniform
+param.nu_nc_norm = 1.0 +/- 0.2
+param.nu_nc_norm.fixed = False
+param.nu_nc_norm.range = nominal + [-.5,+.5]
+
+[utils.hist]
+calc_mode = events
+apply_mode = graphnet_dynamic_binning
+error_method = sumw2""" % (
```

Comment:
This block, like L15-L23 above, are mostly static configurations, and changes are only made in a handful of places. That's why I don't think these should be included as part of `.py` -files. I think it would be natural to package these sections as two separate configuration files (`binning.cfg` and `pipeline.cfg`, respectively), respresenting sensible default settings, rather than defining these files entirely one-the-fly. Then these setting files could be updated, either in-file or programatically, as needed. I have written a bit of code using the `configupdater` that can do this pretty seamlessly. Alternatively, could we use the `param.name = ${osc:name}`-type syntax used elsewhere in these config files?

Reply:
@asogaard thanks. I did look into configparser to handle this, but it requires a special structure in the config files that PISA doesn't support - so I resolved to this. If you have concrete ideas how to handle this, then I'd love to see them.

Its essential that we can edit these config variables on the fly, but I agree that writing the static stuff everytime is unneccesary.

Reply:
Fixed this thanks to snippet from @asogaard